vue-client/src/views/Inspector.vue

```diff
@@ -821,9 +859,46 @@ export default {
       }, 5000);
     },
     async preSaveHook(obj) {
-      await checkAutoShelfControlNumber(obj, this.settings, this.user);
+      await checkAutoShelfControlNumber(obj, this.settings, this.user, this.resources);
+      await this.saveInlined(obj);
+
       return obj;
     },
+    async saveInlined(obj) {
+      // TODO PROPER ERROR HANDLING
+      if (this.inlinedIds.includes(obj['@graph'][1]['@id'])) {
+        // we only handle one level of inlined docs so far
+        return;
+      }
+
+      const inlined = RecordUtil.extractInlinedData(obj['@graph'][1], this.inlinedIds, this.resources);
+      for (const id of this.inlinedIds) {
+        if (inlined[id]) {
+          // FIXME
+          // eslint-disable-next-line no-await-in-loop
```

Comment:
This could probably be rewritten to  `for await (const id of this.inlinedIds) {...` to fix the `no-await-in-loop` issue but I think it works for now!

Reply:
See https://github.com/libris/lxlviewer/pull/1091/commits/eb47f1ee30c2117ce3c90d551b513aa9c4fd1d6a

Reply:
Even better! 👌